src/H5private.h

```diff
@@ -139,6 +139,15 @@
 #define R_OK 04
 #endif
 
+/* uthash is an external, header-only hash table implementation.
```

Comment:
If this is brought in up here, the similar code in H5Ipkg.h and H5Dmpio.c should be removed.

Reply:
Also - I understand that we don't want to redefine "HASH*" in the uthash.h header, but some sense of self-defense here if "HASH*" was already defined in a header that H5private has already included would be good.   Something like "HDcompile_assert(!defined(HASH_FIND));" before including the header?

Reply:
Done (in #1928)

H5Ipkg.h was removed in this PR. I missed H5Dmpio.c.